tests/HTMLPurifierTest.hack

```diff
@@ -5,14 +5,67 @@ namespace HTMLPurifier\_Private\Tests;
 use function Facebook\FBExpect\expect;
 use type Facebook\HackTest\HackTest;
 use namespace HTMLPurifier;
-use namespace HTMLPurifier\{Strategy, Token, Lexer};
+use namespace HTMLPurifier\{Strategy, Token, Lexer, Enums};
 
 class HTMLPurifierTest extends HackTest {
+	private function standardPolicy(): HTMLPurifier\HTMLPurifier_Policy {
+		$policy = HTMLPurifier\HTMLPurifier_Policy::fromEmpty();
+		$policy->addAllowedTags(
+			keyset[
+				Enums\HtmlTags::B,
+				Enums\HtmlTags::UL,
+				Enums\HtmlTags::LI,
+				Enums\HtmlTags::OL,
+				Enums\HtmlTags::H2,
+				Enums\HtmlTags::H4,
+				Enums\HtmlTags::BR,
+				Enums\HtmlTags::DIV,
+				Enums\HtmlTags::STRONG,
+				Enums\HtmlTags::DEL,
+				Enums\HtmlTags::EM,
+				Enums\HtmlTags::PRE,
+				Enums\HtmlTags::CODE,
+				Enums\HtmlTags::TABLE,
+				Enums\HtmlTags::TBODY,
+				Enums\HtmlTags::TD,
+				Enums\HtmlTags::TH,
+				Enums\HtmlTags::THEAD,
+				Enums\HtmlTags::TR,
+			],
+		);
+		$policy->addAllowedTagsWithAttributes(
+			dict[
+				Enums\HtmlTags::A => keyset[
+					Enums\HtmlAttributes::ID,
+					Enums\HtmlAttributes::NAME,
+					Enums\HtmlAttributes::HREF,
+					Enums\HtmlAttributes::TARGET,
+					Enums\HtmlAttributes::REL,
+				],
+				Enums\HtmlTags::H3 => keyset[Enums\HtmlAttributes::CLASSES],
+				Enums\HtmlTags::P => keyset[Enums\HtmlAttributes::CLASSES],
+				Enums\HtmlTags::ASIDE => keyset[Enums\HtmlAttributes::CLASSES],
+				Enums\HtmlTags::IMG => keyset[
+					Enums\HtmlAttributes::SRC,
+					Enums\HtmlAttributes::ALT,
+					Enums\HtmlAttributes::CLASSES,
+					Enums\HtmlAttributes::WIDTH,
+					Enums\HtmlAttributes::HEIGHT,
+					Enums\HtmlAttributes::SRCSET,
+					Enums\HtmlAttributes::SIZES,
+				],
+			],
+		);
+
+		return $policy;
+	}
 
 	public function testMissingEndTags(): void {
 		echo "\nrunning testMissingEndTags()...";
 		//porting over first config classes....
 		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$policy = HTMLPurifier\HTMLPurifier_Policy::fromDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
 
 		$dirty_html = '<b>Bold';
 		$purifier = new HTMLPurifier\HTMLPurifier($config);
```

Comment:
For a lot of these, we clobber the `$purifier` here. Does that mean we don't need to change the way we are defining it (with the `$config` and `$policy` vars passed in)?

Reply:
Ah there were some tests that were passing with the default/original setup. I must have missed some of those, so I will go through and clean those up